opencv/src/OpenCV/ImgProc/StructuralAnalysis.hsc

```diff
@@ -399,3 +400,24 @@ minAreaRect points =
       |]
   where
     c'numPoints = fromIntegral $ V.length points
+
+{- |  Tests a contour convexity.
+
+The function tests whether the input contour is convex or not.
+The contour must be simple, that is, without self-intersections.
+Otherwise, the function output is undefined.
+
+<http://docs.opencv.org/3.0-last-rst/modules/imgproc/doc/structural_analysis_and_shape_descriptors.html?highlight=contourarea#iscontourconvex OpenCV Sphinx doc>
+-}
+isContourConvex
+    :: (IsPoint2 point2 CFloat)
```

Comment:
I prefer `Float` to `CFloat`. `CFLoat` is just a newtype wrapper around `Float` (in GHC), and more convenient for the user.

Looking at the c++ implementation the depth may either be `CV_32F` or `CV_32S`. So I suggest a constraint like:

```haskell
(IsPoint2 point2 depth, depth `In` '[Float, Int32])
```
Having just written this I now realize that this complicates the quasi-quote splice a few lines down. For now just he `Float` type is acceptable. I'll try and implement a general function for converting Haskell vectors of points to OpenCV mats.